src/sysusers/sysusers.c

```diff
@@ -1783,28 +1865,30 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
-        if (optind < argc) {
+        /* If arg_config_name is specified, overrides should be allowed and so we'll
+         * look for config files that would override our configuration (treating the
+         * configuration given on the command as if it was in priority between /run
+         * and /usr/). Otherwise, we look for all files when there are no no
+         * positional arguments. */
+        if (arg_config_name)
+                r = read_config_files(override_file_dirs, arg_config_name);
+        else
+                r = read_config_files(conf_file_dirs, NULL);
```

Comment:
When `--config-name` is not given, then 
1. at first, try to read all .conf files, 
2. and if there are no .conf files in `conf_file_dirs`, then read config files given by the argument.

Is it really expected behavior? I think this breaks the compatibility...